gammapy/time/plot.py

```diff
@@ -2,31 +2,126 @@
+    time_diff = time_diff[0:-1]
+
+    # Midpoints of each bin.
+    time_mid = time_start[0:-1] + time_diff
+
+    flux_history = fermi_cat[1].data[catalog_index]['Flux_History'][time_index_start: time_index_end]
+
+    flux_history_lower_bound = fermi_cat[1].data[catalog_index]['Unc_Flux_History'][time_index_start: time_index_end, 0]
+    flux_history_upper_bound = fermi_cat[1].data[catalog_index]['Unc_Flux_History'][time_index_start: time_index_end, 1]
+    flux_history_lower_bound = abs(flux_history_lower_bound)
+
+    # Change bins with no flux value from nan to zero.
+    for i in range(0, np.size(flux_history_lower_bound)):
+        if math.isnan(flux_history_lower_bound[i]):
+            flux_history_lower_bound[i] = 0
+
+    # Create array of upper limits where no lower bound was recorded.
+    upper_lims_x = np.copy(time_mid)
+
+    upper_lims_y_start = np.copy(flux_history_upper_bound)
+
+    upper_lims_y = np.copy(flux_history)
+
+    upper_lims_x = \
+        upper_lims_x[np.where(flux_history_lower_bound <= 0)]
+
+    upper_lims_y_start = \
+        upper_lims_y_start[np.where(flux_history_lower_bound <= 0)]
+
+    upper_lims_y = \
+        upper_lims_y[np.where(flux_history_lower_bound <= 0)]
+
+    upper_lims_y_end = np.copy(upper_lims_y_start)
+
+    # Scale the size of the arrow to an arbitrary value.
+    upper_lims_y_end *= -0.3
+
+    # Create an array of data points where a lower bound was recorded.
+    time_mid = \
+        time_mid[np.where((flux_history_lower_bound) > 0)]
+
+    time_diff = \
+        time_diff[np.where((flux_history_lower_bound) > 0)]
+
+    flux_history = \
+        flux_history[np.where((flux_history_lower_bound) > 0)]
 
-    if ax is None:
-        ax = plt.gcf()
+    flux_history_upper_bound = \
+        flux_history_upper_bound[np.where((flux_history_lower_bound) > 0)]
 
-    td = time[1:] - time[:-1]
+    flux_history_lower_bound = \
+        flux_history_lower_bound[np.where((flux_history_lower_bound) > 0)]
 
-    # TODO: implement!
-    raise NotImplementedError
+    # Plot data points and upper limits.
+    plt.errorbar(time_mid, flux_history, yerr=(flux_history_lower_bound, flux_history_upper_bound), \
+                 xerr=0.5 * time_diff, \
+                 marker='o', elinewidth=1, linewidth=0, color='black')
+    plt.errorbar(upper_lims_x[np.where(upper_lims_y > 0)], upper_lims_y[np.where(upper_lims_y > 0)], \
+                 yerr=(upper_lims_y_end[np.where(upper_lims_y > 0)], upper_lims_y_start[np.where(upper_lims_y > 0)]), \
+                 marker='o', elinewidth=1, linewidth=0, lolims=True, color='black')
+    plt.errorbar(upper_lims_x[np.where(upper_lims_y <= 0)], upper_lims_y[np.where(upper_lims_y <= 0)], \
+                 yerr=(upper_lims_y_end[np.where(upper_lims_y <= 0)], upper_lims_y_start[np.where(upper_lims_y <= 0)]), \
+                 marker=None, elinewidth=1, linewidth=0, lolims=True, color='black')
+    plt.xlabel('time [Fermi MET]')
```

Comment:
I would much prefer to get a time axis with years and maybe months as labels than Fermi MET.
Those numbers mean nothing to me and probably to most other astronomers as well.

Here's some info how to convert times to a format matplotlib understands:
http://astropy.readthedocs.org/en/latest/time/index.html?highlight=matplotlib#get-representation

This will be the hard part of this PR ... figuring out how to correctly create `astropy.timeTime` objects and how to correctly plot them with matplotlib.
If you have any questions or get stuck, please ask and I can investigate how to do this also.